modules/renter/download.go

```diff
@@ -45,6 +45,8 @@ func (hf *hostFetcher) pieces(chunk uint64) []pieceData {
 
 // fetch downloads the piece specified by p.
 func (hf *hostFetcher) fetch(p pieceData) ([]byte, error) {
+	hf.conn.SetDeadline(time.Now().Add(1 * time.Minute)) // sufficient to transfer 4 MB over 500 kbps
```

Comment:
500kbps is too optimistic. Especially considering hosts might be getting multiple pieces at the same time.

Is there a way to extend the deadline? Maybe you could extend the deadline by 3 minutes once the uploading starts?

I'm not sure what our average host connection is. But I know a lot of homes have only 1mbps up or 0.5 mbps up. I don't think that there are many that are too far below that, but multiple simultaneous connections might still cause issues.